### PR TITLE
Implement results from no timestamp in Explorer

### PR DESCRIPTION
**Pull request by paulstn**
### Description
To allow for querying of flint materialized views and covering indices which may not have timestamps, Explorer is being extended to show results from queries with no timestamps
Changes:

- Check field mapping for no timestamp
- Send out query without timestamp and get results to Explorer
- Disable timepicker
- Get rid of time column in data grid
- Saved query: modify load
- Visualization changes
- Disable patterns
- Enable Hit Counter

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by paulstn** [comment]
https://github.com/opensearch-project/dashboards-observability/issues/1558
Will create tests later, but it will take some time since I would need to custom define an index with no timestamp on top of creating all of the tests.

**Comment by mengweieric** [comment]
did you test with saved object?

**Comment by mengweieric** [comment]
these are not small changes, could we add tests?

File: public/components/common/search/date_picker.tsx [hunk 1]
```diff
@@ -23,28 +24,40 @@ export function DatePicker(props: IDatePickerProps) {
...
-  const allowTimeChanging = !coreRefs.queryAssistEnabled || isAppAnalytics;
+
+  let setStartTime;
+  let setEndTime;
+  let setDisabled;
+  let toolTipMessage;
+
+  if (coreRefs.queryAssistEnabled && !isAppAnalytics) {
+    // is query assistant inside log explorer
+    setStartTime = QUERY_ASSIST_START_TIME;
```
[the remark this answers is not in the document]
Comment (paulstn) [reply public/components/common/search/date_picker.tsx:35]: I meant set in the same way as 'the set time' but I can see the confusion, I'll use something different.

File: public/components/common/search/search.tsx [hunk 2]
```diff
@@ -409,33 +410,35 @@ export const Search = (props: any) => {
...
+                    startTime={startTime}
+                    endTime={endTime}
+                    setStartTime={setStartTime}
+                    setEndTime={setEndTime}
+                    setIsOutputStale={setIsOutputStale}
+                    liveStreamChecked={props.liveStreamChecked}
+                    onLiveStreamChange={props.onLiveStreamChange}
+                    handleTimePickerChange={(tRange: string[]) => {
+                      // modifies run button to look like the update button, if there is a time change, disables timepicker setting update if timepicker is disabled
+                      setNeedsUpdate(
```
[the remark this answers is not in the document]
Comment (paulstn) [reply public/components/common/search/search.tsx:426]: My PR didn't introduce this, and it seems like this name refers to the 'update' button. It seems like this method is used in many different files, would you like me to change it now in this PR?

[the remark this answers is not in the document]
Comment (mengweieric) [reply public/components/common/search/search.tsx:426]: You may create an issue listing all these items that need a followups, we can follow up on this later thanks

**Review by Swiddis — APPROVED** [review]

**Review by mengweieric — APPROVED** [review]

**Pull request merged**